src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,138 @@
+        return SphereClient.of(clientConfig, httpClient, tokenSupplier);
+    }
+
+    protected static SphereClient decorateSphereClient(
+            @Nonnull final SphereClient underlyingClient,
+            @Nonnull final long maxRetryAttempt,
+            @Nonnull final Function<RetryContext, Duration> durationFunction,
+            @Nonnull final int maxParallelRequests) {
+
+        final SphereClient retryClient = withRetry(underlyingClient, maxRetryAttempt, durationFunction);
+        return withLimitedParallelRequests(retryClient, maxParallelRequests);
+    }
+
+        /**
+         * Gets an asynchronous {@link HttpClient} to be used by the {@link BlockingSphereClient}.
+         *
+         * @return {@link HttpClient}
+         */
+    private static HttpClient getHttpClient() {
+        final AsyncHttpClient asyncHttpClient =
+                new DefaultAsyncHttpClient(
+                        new DefaultAsyncHttpClientConfig.Builder().build());
+        return AsyncHttpClientAdapter.of(asyncHttpClient);
+    }
+
+    private static SphereClient withRetry(
+            @Nonnull final SphereClient delegate,
+            long maxRetryAttempt,
+            @Nonnull final Function<RetryContext, Duration> durationFunction) {
+
+        final RetryAction scheduledRetry = RetryAction.ofScheduledRetry(maxRetryAttempt, durationFunction);
+        final RetryPredicate http5xxMatcher = RetryPredicate.ofMatchingStatusCodes(errCode -> errCode > 499 && errCode < 600);
+        final List<RetryRule> retryRules = Collections.singletonList(RetryRule.of(http5xxMatcher, scheduledRetry));
+        return RetrySphereClientDecorator.of(delegate, retryRules);
+    }
+
+    /**
+     * Computes a exponential backoff time delay in milliseconds to be used in retries, the delay grows with failed
+     * retry attempts count with a randomness interval.
+     * (see: <a href=https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/>)
+     * (see: <a href="http://dthain.blogspot.com/2009/02/exponential-backoff-in-distributed.html"/>)
+     *
+     * @param retryAttempt the number of attempts already tried by the client.
+     * @return a duration in milliseconds, that grows with the number of failed attempts.
+     */
+    protected static Duration calculateDurationWithExponentialRandomBackoff(final long retryAttempt,
+                                                                            long initialRetryDelay, long timeout) {
+        final double exponentialFactor = Math.pow(2, retryAttempt - 1);
+        final double jitter = 1 + Math.random();
+        final long delay = (long)Math.min(initialRetryDelay * exponentialFactor * jitter, timeout);
+        return Duration.ofMillis(delay);
+    }
+
+    private static SphereClient withLimitedParallelRequests(final SphereClient delegate, final int maxParallelRequests) {
+        return QueueSphereClientDecorator.of(delegate, maxParallelRequests);
+    }
+
```

Comment:
The formatting could be nicer for the class, some empty spaces and intentions for example here.